src/functionalTest/kotlin/kotlinx/kover/test/functional/core/Runner.kt

```diff
@@ -0,0 +1,308 @@
+    }
+
+    override fun kover(rootExtensionScript: String) {
+        koverConfigs += rootExtensionScript
+    }
+
+    override fun sources(template: String): ProjectRunner {
+        fun File.processDir(result: MutableMap<String, String>, path: String = "") {
+            listFiles()?.forEach { file ->
+                val filePath = "$path/${file.name}"
+                if (file.isDirectory) {
+                    file.processDir(result, filePath)
+                } else if (file.exists() && file.length() > 0) {
+                    result += filePath to file.readText()
+                }
+            }
+        }
+
+        File(SOURCES_PATH, "$template/main").processDir(mainSources)
+        File(SOURCES_PATH, "$template/test").processDir(testSources)
+        return this
+    }
+
+    override fun verification(rules: Iterable<VerificationRule>): ProjectRunner {
+        this.rules += rules
+        return this
+    }
+
+    override fun config(script: String): ProjectRunner {
+        kotlinScripts += script
+        groovyScripts += script
+        return this
+    }
+
+    override fun config(kotlin: String, groovy: String): ProjectRunner {
+        kotlinScripts += kotlin
+        groovyScripts += groovy
+        return this
+    }
+
+    override fun check(vararg args: String, block: RunResult.() -> Unit) {
+        setDefaults()
+        initialized = true
+
+        languages.forEach { language ->
+            types.forEach { type ->
+                engines.forEach { engine ->
+                    createProject(language, type, engine).run(listOf(*args), block)
+                }
+            }
+        }
+    }
+
+    private fun setDefaults() {
+        if (languages.isEmpty()) {
+            languages += GradleScriptLanguage.KOTLIN
+        }
+        if (types.isEmpty()) {
+            types += ProjectType.KOTLIN_JVM
+        }
+        if (engines.isEmpty()) {
+            engines += null
+        }
+        if (pluginVersion == null) {
+            pluginVersion = "0.4.2" // TODO read from properties
+        }
+    }
+
+    private fun File.run(args: List<String>, block: RunResult.() -> Unit) {
+        val buildResult = GradleRunner.create()
+            .withProjectDir(this)
+            .withPluginClasspath()
+            .addPluginTestRuntimeClasspath()
+            .withArguments(args)
+            .build()
+
+        RunResult(buildResult, this).apply(block)
+    }
+
+    private fun createProject(language: GradleScriptLanguage, type: ProjectType, engine: CoverageEngine?): File {
+        val projectDir = File(rootDir, System.currentTimeMillis().toString()).also { it.mkdirs() }
+
+        val extension = if (language == GradleScriptLanguage.KOTLIN) "gradle.kts" else "gradle"
+
+        File(projectDir, "build.$extension").writeText(
+            scriptTemplate(true, language, type)
+                .replace("//PLUGIN_VERSION", pluginVersion!!)
+                .replace("//REPOSITORIES", "")
```

Comment:
What this replacement is for?

Reply:
Currently not used, but for the android project type, additional dependencies and repositories will be needed.

Reply:
Thanks